istioctl/pkg/injector/injector-list.go

```diff
@@ -229,19 +229,23 @@ func getInjectedRevision(namespace *corev1.Namespace, hooks []admitv1.MutatingWe
 }
 
 func getMatchingNamespaces(hook *admitv1.MutatingWebhookConfiguration, namespaces []corev1.Namespace) []corev1.Namespace {
-	retval := make([]corev1.Namespace, 0)
+	matched := map[string]corev1.Namespace{}
 	for _, webhook := range hook.Webhooks {
 		nsSelector, err := metav1.LabelSelectorAsSelector(webhook.NamespaceSelector)
 		if err != nil {
-			return retval
+			return nil
 		}
 
 		for _, namespace := range namespaces {
 			if nsSelector.Matches(api_pkg_labels.Set(namespace.Labels)) {
-				retval = append(retval, namespace)
+				matched[namespace.Name] = namespace
```

Comment:
I think it might be better to use `set` to prevent duplication.

Reply:
Sounds good, fixed